src/main/java/com/iota/iri/zmq/TimeWindowedApproveeCounter.java

```diff
@@ -0,0 +1,93 @@
+package com.iota.iri.zmq;
+
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.dag.DAGHelper;
+import com.iota.iri.utils.dag.TraversalException;
+
+import java.util.HashSet;
+
+/**
+ * Utility class to count the number of distinct approvees that have an arrival time within the given time window.
+ */
+class TimeWindowedApproveeCounter {
+
+    private final Tangle tangle;
+    private final long minTransactionAgeSeconds;
+    private final long maxTransactionAgeSeconds;
+
+    TimeWindowedApproveeCounter(Tangle tangle, long minTransactionAgeSeconds, long maxTransactionAgeSeconds) {
+        this.tangle = tangle;
+        this.minTransactionAgeSeconds = minTransactionAgeSeconds;
+        this.maxTransactionAgeSeconds = maxTransactionAgeSeconds;
+    }
+
+    private boolean continueTraversal(long now, TransactionViewModel transactionViewModel) {
+        // stop the traversing as soon as the transaction is older than the max
+        final long age = now - transactionViewModel.getArrivalTime();
+        return age <= maxTransactionAgeSeconds;
+    }
+
+    /**
+     * Checks whether the transaction is in the validity time window.
+     * 
+     * @param now         current time epoch in seconds
+     * @param transaction transaction to check
+     * @return true, if the arrival time is in the time window, false otherwise
+     */
+    boolean isInTimeWindow(long now, TransactionViewModel transaction) {
+
+        final long age = now - transaction.getArrivalTime();
+        return (age >= minTransactionAgeSeconds && age <= maxTransactionAgeSeconds);
+    }
+
+    /**
+     * Counts the number of distinct approvees of the starting transaction, that are within the validity time window.
+     * 
+     * The approvee traversal is not continued for transactions that are older than the maximum age.
+     *
+     * @param now                     current time epoch in seconds
+     * @param startingTransactionHash the starting point of the traversal
+     * @param processedTransactions   a set of hashes that shall be considered as "processed" already and that will
```

Comment:
`ignoredTransactions`?

Reply:
Not sure about this, but it was tricky for me to parse

Reply:
Clarified the doc